### DISCUSSION TITLE
How to Ignore Array fields

### DISCUSSION THREAD

**Discussion by BledItIt** [bellrichm/WeeWX-MQTTSubscribe#218] [Q&A]
My log is filled with:
ERROR user.MQTTSubscribe: (Driver) Skipping hlxt3 because data is an array and has no configured subfields. Array=['---', '---', '---', '---', '---', '---', '86', '86']

How can I ignore those array fields? (do not need them) If they were marked DEBUG the logfile would not fill up.

It ignores non array fields just fine and marks them as debug if they are not noted in the weewx.conf

--------------------------------------------

[MQTTSubscribeDriver]
    # This section is for the MQTTSubscribe driver.

    # The driver to use:
    driver = user.MQTTSubscribe

    # The MQTT server.
    # Default is localhost.
    host = ##.##.##.##

    # The port to connect to.
    # Default is 1883.
    port = 1883

    # Maximum period in seconds allowed between communications with the broker.
    # Default is 60.
    keepalive = 60

    # username for broker authentication.
    # Default is None.
    username = #######

    # password for broker authentication.
    # Default is None.
    password = #######

    # Configuration for the message callback.
    [[message_callback]]
        # The format of the MQTT payload.
        # Currently support: individual, json, keyword
        # Must be specified.
        type = json

    # The topics to subscribe to.
    [[topics]]

        # Units for MQTT payloads without unit value.
        # Valid values: US, METRIC, METRICWX
        # Default is: US
        unit_system = US

        [[[/wflexp.json]]]
            ignore = true    # opt out of processing fields for this topic by default - override per field
            [[[[tempout]]]]
                ignore = false    # opt back in for this field
                name = outTemp
            [[[[tempin]]]]
                ignore = true    # opt back in for this field
                name = inTemp
            [[[[humin]]]]
                ignore = true    # opt back in for this field
                name = inHumidity
            [[[[humout]]]]
                ignore = false    # opt back in for this field
                name = outHumidity
            [[[[windspd]]]]
                ignore = false    # opt back in for this field
                name = windSpeed
            [[[[winddir]]]]
                ignore = false    # opt back in for this field
                name = windDir
            [[[[gust]]]]
                ignore = false    # opt back in for this field
                name = windGust
            [[[[gustdir]]]]
                ignore = false    # opt back in for this field
                name = windGustDir
            [[[[bar]]]]
                ignore = false    # opt back in for this field
                name = barometer
            [[[[rainyear]]]]
                ignore = false    # opt back in for this field
                contains_total = true
                name = rain
            [[[[rainr]]]]
                ignore = false    # opt back in for this field
                name = rainRate
            [[[[etyear]]]]
                ignore = true    # opt back in for this field
                name = ET
                contains_total = true
            [[[[dew]]]]
                ignore = false    # opt back in for this field
                name = dewpoint
            [[[[heat]]]]
                ignore = false    # opt back in for this field
                name = heatindex
            [[[[solar]]]]
                ignore = true    # opt back in for this field
                name = radiation
            [[[[uv]]]]
                ignore = true    # opt back in for this field
                name = UV
                conversion_error_to_none = true
            [[[[bat]]]]
                ignore = false    # opt back in for this field
                name = consBatteryVoltage
--------------------------------------------------------------------
wflexpj.json 
--------------------------------------------------------------------

wflexpj.json, QOS: 0, retain: 0, payload: b'{"stnname":"########## - ######","stnmod":17,"ver":3.00,"mac":"##############","apmac":"############","ip":"##########","ssid":"#########","rssi":-68,"wifimod":0,"lastboot":1717000260,"uptime":9725566,"wflver":"2.43","loctime":1726707610,"utctime":1726722010,"tzone":10,"units":4,"conlati":####,"conlongi":#####,"wfllati":##.####,"wfllongi":-##.########,"tempout":62.7,"humout":87,"tempin":65.0,"humin":61,"windspd":0.0,"winddir":86,"windavg2":0.0,"windavg10":0.0,"gust":0.0,"gustdir":360,"bar":29.916,"bartr":0,"dew":59,"cdew":58.8,"chill":63,"heat":63,"thsw":null,"uv":null,"solar":null,"rainr":0.00000,"storm":5.01000,"rain15":0.00000,"rain1h":0.00000,"raind":0.00000,"rain24":0.20000,"rainmon":6.29000,"rainyear":33.77000,"etday":0.000,"etmon":0.00,"etyear":0.00,"xt":[null,null,null,null,null,null,null],"xlt":[null,null,null,null],"xst":[null,null,null,null],"xh":[null,null,null,null,null,null,null],"xsm":[null,null,null,null],"xlw":[null,null,null,0],"bat":4.84,"trbat":2,"foreico":6,"forrule":45,"sunrt":"7:15","sunst":"19:29","hlbar":[29.915,29.922,"0:30","0:01",30.350,29.832,30.749,29.256],"hlwind":[null,2.0,null,"0:04",19.0,null,28.0,null],"hltempin":[65.0,65.1,"0:35","0:00",77.6,63.6,79.2,44.0],"hlhumin":[61,61,"0:00","0:00",63,42,63,29],"hltempout":[62.6,62.9,"0:58","0:00",83.5,52.2,94.7,7.2],"hlhumout":[87,87,"0:00","0:00",87,39,87,0],"hldew":[59,59,"0:00","0:00",69,40,76,-120],"hlchil":[63,null,"0:00",null,null,52,null,52],"hlheat":[null,64,null,"0:00",87,null,100,null],"hlthsw":[null,null,null,null,null,null,null,null],"hlsolar":[null,0,null,null,0,null,0,null],"hluv":[null,0.0,null,null,0.0,null,0.0,null],"hlrainr":[0.000,0.000,null,null,1.850,null,5.820,null],"hlxt0":[null,null,null,null,null,null,null,null],"hlxt1":[null,null,null,null,null,null,null,null],"hlxt2":[null,null,null,null,null,null,null,null],"hlxt3":[null,null,null,null,null,null,null,null],"hlxt4":[null,null,null,null,null,null,null,null],"hlxt5":[null,null,null,null,null,null,null,null],"hlxt6":[null,null,null,null,null,null,null,null],"hlxh0":[null,null,null,null,null,null,null,null],"hlxh1":[null,null,null,null,null,null,null,null],"hlxh2":[null,null,null,null,null,null,null,null],"hlxh3":[null,null,null,null,null,null,null,null],"hlxh4":[null,null,null,null,null,null,null,null],"hlxh5":[null,null,null,null,null,null,null,null],"hlxh6":[null,null,null,null,null,null,null,null],"hlxst0":[null,null,null,null,null,null,null,null],"hlxst1":[null,null,null,null,null,null,null,null],"hlxst2":[null,null,null,null,null,null,null,null],"hlxst3":[null,null,null,null,null,null,null,null],"hlxlt0":[null,null,null,null,null,null,null,null],"hlxlt1":[null,null,null,null,null,null,null,null],"hlxlt2":[null,null,null,null,null,null,null,null],"hlxlt3":[null,null,null,null,null,null,null,null],"hlxsm0":[null,null,null,null,null,null,null,null],"hlxsm1":[null,null,null,null,null,null,null,null],"hlxsm2":[null,null,null,null,null,null,null,null],"hlxsm3":[null,null,null,null,null,null,null,null],"hlxlw0":[null,null,null,null,null,null,null,null],"hlxlw1":[null,null,null,null,null,null,null,null],"hlxlw2":[null,null,null,null,null,null,null,null],"hlxlw3":[null,null,null,null,null,null,null,null]}'

**Comment by BledItIt** [comment]
what I ended up doing is modifying: (not sure if there would have been anything I could have put in the conf file or not).
If this was the the answer you can close this... 

MQTTSubscribe.py
line 1407 and changed 

            self.logger.error("Skipping %s because data is an array and has no configured subfields. Array=%s" % (new_key, value))
            
to 

            self.logger.debug("Skipping %s because data is an array and has no configured subfields. Array=%s" % (new_key, value))

**Comment by bellrichm** [comment]
more readable payload is attached
[t.json](https://github.com/user-attachments/files/17065323/t.json)

**Comment by bellrichm** [comment]
Arrays should inherit the ignore setting. An issue has been created.
